pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -752,3 +738,17 @@ func makeGvk(kind string) schema.GroupVersionKind {
 		Kind:    kind,
 	}
 }
+
+func shouldGenerateVAP(ct *templates.ConstraintTemplate, generateVAPDefault bool) (bool, error) {
+	source, err := pSchema.GetSourceFromTemplate(ct)
+	if errors.Is(err, pSchema.ErrCodeNotDefined) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	if source.GenerateVAP == nil {
+		return generateVAPDefault, nil
```

Comment:
Should we also check if source has `validations` instead of `rego` here in case generateVAPDefault is true to reduce calls to the vap generation for rego policies?

Reply:
Recall users can also just add one engine with the rego source:
https://github.com/open-policy-agent/gatekeeper-library/pull/519/files#diff-1f24aea6b79b21ecbbc966644df5f8afeea967ff3627a6e505e0a9262a874c8eR45-R47

and
https://open-policy-agent.github.io/gatekeeper-library/website/validation/requiredlabels

Reply:
I thought `GetSourceFromTemplate` was a generic thing, not specific to the cel engine.